src/saltext/vmware/utils/vmc_vcenter_request.py

```diff
@@ -0,0 +1,144 @@
+"""
+    VMC vCenter API Request Module
+"""
+import logging
+
+import requests
+from requests.auth import HTTPBasicAuth
+from requests.exceptions import HTTPError
+from requests.exceptions import RequestException
+from requests.exceptions import SSLError
+from saltext.vmware.utils import vmc_constants
+from saltext.vmware.utils import vmc_request
+
+log = logging.getLogger(__name__)
+
+
+def get_api_session_id(hostname, username, password):
+    """
+    This function returns api_session_id required to perform operations for VMC vCenter.
+
+    hostname
+        Hostname of the vCenter console
+
+    username
+        username required to login to vCenter console
+
+    password
+        password required to login to vCenter console
+
+    """
+    url = vmc_request.set_base_url(hostname) + vmc_constants.VCENTER_API_SESSION_URL
+    headers = {vmc_constants.CONTENT_TYPE: vmc_constants.APPLICATION_JSON}
+    response = requests.post(url, headers=headers, auth=HTTPBasicAuth(username, password))
+    return response.json()
+
+
+def get_headers(hostname, username, password):
+    """
+    This function returns HTTP headers required to perform operations for VMC vCenter.
+
+    hostname
+        Hostname of the vCenter console
+
+    username
+        username required to login to vCenter console
+
+    password
+        password required to login to vCenter console
+
+    """
+    api_session_id = get_api_session_id(hostname, username, password)
+    return {
+        vmc_constants.CONTENT_TYPE: vmc_constants.APPLICATION_JSON,
+        vmc_constants.VMWARE_API_SESSION_ID: api_session_id,
+    }
+
+
+def call_api(
```

Comment:
**suggestion** this appears to be a near/exact duplicate to `vmc_request`'s `call_api`. Is there a good reason not to have a common function for calling the API?

Reply:
Headers part is different for VMC and vcenter APIs.  In order to keep the code simple and understandable, we kept it as separate.